pool/staticpool.go

```diff
@@ -13,35 +14,51 @@ var _ Pool = &StaticPool{}
 // StaticPool is a dummy implementation of a pool service that always returns
 // from the same set of host nodes. It does not do any signature checking.
 type StaticPool struct {
-	Nodes             []store.Node
```

Comment:
This was another example where it was too easy to accidentally have side effects/mutations on the internal slice, messing up some test scenarios.